src/Microsoft.Diagnostics.Monitoring.EventPipe/Traces/ActivityData.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    internal readonly struct ActivityData
```

Comment:
This is a very large struct object. Perhaps a class or record is more suitable? I have not looked at usage yet.

Reply:
It is a super big struct! But that doesn't concern me. It is handled correctly, passed by reference (`out` or `in` in our cases). What this code is trying to do is not create unneeded GC pressure in the monitoring app. It doesn't have to do that, but that was my thinking 😄